Src/GoogleApis.Core/Apis/Requests/Parameters/ParameterUtils.cs

```diff
@@ -121,7 +121,17 @@ private static void IterateParameters(object request, Action<RequestParameterTyp
                 // Call action with the type name and value.
                 if (propertyType.IsValueType || value != null)
                 {
-                    action(attribute.Type, name, value);
+                    if (attribute.Type == RequestParameterType.UserDefiniedQueries)
+                    {
+                        foreach (var pair in (KeyValuePair<string, string>[])value)

```

Comment:
It is worth to have a type validation check, and log if it isn't (I mean the cast for KeyValuePair array)